tutorials/hello/hello_client/src/app/plugins/hello_info/hello_info_plugin.ts

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable prefer-destructuring */
+/* eslint-disable no-loop-func */
+/* eslint-disable @typescript-eslint/prefer-for-of */
+/* eslint-disable for-direction */
+/* eslint-disable no-param-reassign */
+/* eslint-disable @typescript-eslint/no-floating-promises */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+/* eslint-disable consistent-return */
+/* eslint-disable no-useless-return */
+/* eslint-disable import/namespace */
+/* eslint-disable @typescript-eslint/prefer-readonly */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable dot-notation */
+/* eslint-disable @typescript-eslint/require-await */
+/* eslint-disable no-console */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable spaced-comment */
+/* eslint-disable prefer-const */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable class-methods-use-this */
+/* eslint-disable  @typescript-eslint/no-empty-function */
+import { BasePlugin, db as liskDB, PluginInitContext } from 'lisk-sdk';
+import {
+	getDBInstance,
+	setLastCounter,
+	getLastCounter,
+} from './db';
+import { configSchema } from './schemas';
+import { HelloInfoPluginConfig } from './types';
+import { Endpoint } from './endpoint';
+
+export class HelloInfoPlugin extends BasePlugin<HelloInfoPluginConfig> {
+	public configSchema = configSchema;
+	public endpoint = new Endpoint();
+	public counter = 0;
+	public height = 0;
+	private _addressPluginDB!: liskDB.Database;
+
+	public get nodeModulePath(): string {
+		return __filename;
+	}
+
+	public async init(context: PluginInitContext): Promise<void> {
+		await super.init(context);
+	}
+
+	// loads DB instances and initiates counter for first run.
+	public async load(): Promise<void> {
+		if (this.config.enablePlugin) {
```

Comment:
I think this is a bit misleading, as the plugin will be loaded in memory anyways even though this part of code is not executed. I think we can remove enablePlugin flag from the config and other places, we simply don't call `app.registerPlugin(new HelloInfoPlugin());`when the user doesn't want to load plugin

Reply:
Ummmmm! the idea is to have a use case for passing config to a plugin. Can you please suggest any other valid and simple use case for the config of a plugin that I can add?

Reply:
The old config usecase has been updated.

I have added a new `syncInterval` config property, allowing my plugin to sync after an interval. The plugin will now store any newly generated **HelloEvents** in the plugin's database by checking after a specific interval.